cmd/state-transition/state_transition.cpp

```diff
@@ -0,0 +1,318 @@
+        block.header.difficulty = intx::from_string<intx::uint256>(get_env("currentDifficulty"));
+    }
+
+    if (contains_env("currentBaseFee") && rev >= EVMC_LONDON) {
+        block.header.base_fee_per_gas = intx::from_string<intx::uint256>(get_env("currentBaseFee"));
+    }
+
+    if (rev >= EVMC_SHANGHAI) {
+        block.withdrawals = std::vector<Withdrawal>{};
+        block.header.withdrawals_root = kEmptyRoot;
+    }
+
+    block.header.transactions_root = protocol::compute_transaction_root(block);
+    block.header.ommers_hash = kEmptyListHash;
+
+    auto parent_block = silkworm::Block();
+    parent_block.header.gas_limit = block.header.gas_limit;
+    parent_block.header.gas_used = parent_block.header.gas_limit / protocol::kElasticityMultiplier;
+    parent_block.header.number = block.header.number - 1;
+    parent_block.header.base_fee_per_gas = block.header.base_fee_per_gas;
+    state.insert_block(parent_block, block.header.parent_hash);
+
+    return block;
+}
+
+std::unique_ptr<silkworm::InMemoryState> StateTransition::get_state() {
+    auto state = std::make_unique<silkworm::InMemoryState>();
+
+    auto pre = test_data_["pre"];
+
+    for (const auto& preState : pre.items()) {
+        const auto address = to_evmc_address(preState.key());
+        const nlohmann::json preStateValue = preState.value();
+
+        auto account = silkworm::Account();
+        account.balance = intx::from_string<intx::uint256>(preStateValue.at("balance"));
+        account.nonce = std::stoull(std::string(preStateValue.at("nonce")), nullptr, 16);
+
+        const Bytes code{from_hex(std::string(preStateValue.at("code"))).value()};
+        account.code_hash = silkworm::bit_cast<evmc_bytes32>(keccak256(code));
+        account.incarnation = kDefaultIncarnation;
+
+        state->update_account(address, /*initial=*/std::nullopt, account);
+        state->update_account_code(address, account.incarnation, account.code_hash, code);
+
+        for (const auto& storage : preStateValue.at("storage").items()) {
+            Bytes key{from_hex(storage.key()).value()};
+            Bytes value{from_hex(storage.value().get<std::string>()).value()};
+            state->update_storage(address, account.incarnation, to_bytes32(key), /*initial=*/{}, to_bytes32(value));
+        }
+    }
+
+    return state;
+}
+
+std::unique_ptr<evmc::address> StateTransition::private_key_to_address(const std::string& private_key) {
+    /// Example
+    // private key: 0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8
+    // public key : 043a514176466fa815ed481ffad09110a2d344f6c9b78c1d14afc351c3a51be33d8072e77939dc03ba44790779b7a1025baf3003f6732430e20cd9b76d953391b3
+    // address    : 0xa94f5374Fce5edBC8E2a8697C15331677e6EbF0B
+
+    auto private_key_bytes = from_hex(private_key).value();
+
+    silkworm::sentry::common::EccKeyPair pair = silkworm::sentry::common::EccKeyPair(private_key_bytes);
+
+    uint8_t out[20];
```

Comment:
You can replace 20 with `kAddressLength` constant defined in `silkworm/core/common/base.hpp`